pinot-core/src/main/java/org/apache/pinot/core/data/recordtransformer/PinotDataType.java

```diff
@@ -632,4 +721,22 @@ public static PinotDataType getPinotDataType(FieldSpec fieldSpec) {
             "Unsupported data type: " + dataType + " in field: " + fieldSpec.getName());
     }
   }
+
+  public static PinotDataType getSpecificMapDataTypeFromMap(Map<Object, Object> map) {
+    Iterator<Object> iterator = map.values().iterator();
+    Object obj = iterator.next();
+    if (obj instanceof Integer) {
+      return PinotDataType.INTEGER_MAP;
+    } else if (obj instanceof Long) {
+      return PinotDataType.LONG_MAP;
+    } else if (obj instanceof Float) {
+      return PinotDataType.FLOAT_MAP;
+    } else if (obj instanceof Double) {
+      return PinotDataType.DOUBLE_MAP;
+    } else if (obj instanceof String) {
+      return PinotDataType.STRING_MAP;
+    } else {
+      throw new IllegalStateException(String.format("'%s' isn't supported in the hash map.", obj.getClass()));
```

Comment:
Since we don't completely support complex map cases like nested map and single-entry map with multi-valued value, we can add the logic to it once it's supported. We can add a TODO here.